Signal/src/call/CallService.swift

```diff
@@ -1495,6 +1534,18 @@ private class SignalCallData: NSObject {
                 // call failed before any call record could be created, make one now.
                 handleMissedCall(failedCall)
             }
+
+            // Ensure there is a call record.
+            let callType = (failedCall.direction == .incoming
```

Comment:
We weren't always creating a call record, e.g. if caller hung up while ringing.

Reply:
The case you've described should be covered by our existing handleRemoteHangup logic, what am I missing?

Reply:
Yup - we've just missed cases before and so I thought it'd be conservative to make sure we always leave a call record, given all of the edge cases in calling.  If you'd rather remove, we can.